<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Russell Bainer" />

<meta name="date" content="2020-05-12" />

<title>Example Workflow For Processing a CRISPR-based Pooled Screen</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Example Workflow For Processing a CRISPR-based Pooled Screen</h1>
<h4 class="author">Russell Bainer</h4>
<h4 class="date">2020-05-12</h4>



<div id="example-workflow" class="section level3">
<h3>Example Workflow</h3>
<p>This is an example workflow for processing a pooled CRISPR-based screen using the provided sample data. See the various manpages for additional visualization options and algorithmic details.</p>
<p>Load dependencies and data</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(Biobase)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">library</span>(limma)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="kw">library</span>(gCrisprTools)</span>
<span id="cb1-4"><a href="#cb1-4"></a></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="kw">data</span>(<span class="st">&quot;es&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="kw">data</span>(<span class="st">&quot;ann&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="kw">data</span>(<span class="st">&quot;aln&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</span></code></pre></div>
<p>Make a sample key, structured as a factor with control samples in the first level</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>sk &lt;-<span class="st"> </span><span class="kw">relevel</span>(<span class="kw">as.factor</span>(<span class="kw">pData</span>(es)<span class="op">$</span>TREATMENT_NAME), <span class="st">&quot;ControlReference&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw">names</span>(sk) &lt;-<span class="st"> </span><span class="kw">row.names</span>(<span class="kw">pData</span>(es))</span></code></pre></div>
<p>Generate a contrast of interest using voom/limma; pairing replicates is a good idea if that information is available.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span><span class="st"> </span><span class="dv">0</span> <span class="op">+</span><span class="st"> </span>REPLICATE_POOL <span class="op">+</span><span class="st"> </span>TREATMENT_NAME, <span class="kw">pData</span>(es))</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">colnames</span>(design) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&#39;TREATMENT_NAME&#39;</span>, <span class="st">&#39;&#39;</span>, <span class="kw">colnames</span>(design))</span>
<span id="cb3-3"><a href="#cb3-3"></a>contrasts &lt;-<span class="kw">makeContrasts</span>(DeathExpansion <span class="op">-</span><span class="st"> </span>ControlExpansion, <span class="dt">levels =</span> design)</span></code></pre></div>
<p>Optionally, trim of trace reads from the unnormalized object (see man page for details)</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>es &lt;-<span class="st"> </span><span class="kw">ct.filterReads</span>(es, <span class="dt">trim =</span> <span class="dv">1000</span>, <span class="dt">sampleKey =</span> sk)</span></code></pre></div>
<p>Normalize, convert to a voom object, and generate a contrast</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>es &lt;-<span class="st"> </span><span class="kw">ct.normalizeGuides</span>(es, <span class="dt">method =</span> <span class="st">&quot;scale&quot;</span>, <span class="dt">plot.it =</span> <span class="ot">TRUE</span>) <span class="co">#See man page for other options</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>vm &lt;-<span class="st"> </span><span class="kw">voom</span>(<span class="kw">exprs</span>(es), design)</span>
<span id="cb5-3"><a href="#cb5-3"></a></span>
<span id="cb5-4"><a href="#cb5-4"></a>fit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(vm, design)</span>
<span id="cb5-5"><a href="#cb5-5"></a>fit &lt;-<span class="st"> </span><span class="kw">contrasts.fit</span>(fit, contrasts)</span>
<span id="cb5-6"><a href="#cb5-6"></a>fit &lt;-<span class="st"> </span><span class="kw">eBayes</span>(fit)</span></code></pre></div>
<p>Edit the annotation file if you used <code>ct.filterReads</code> above</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>ann &lt;-<span class="st"> </span><span class="kw">ct.prepareAnnotation</span>(ann, fit, <span class="dt">controls =</span> <span class="st">&quot;NoTarget&quot;</span>)</span></code></pre></div>
<p>Summarize gRNA signals to identify target genes of interest</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>resultsDF &lt;-</span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="st">  </span><span class="kw">ct.generateResults</span>(</span>
<span id="cb7-3"><a href="#cb7-3"></a>    fit,</span>
<span id="cb7-4"><a href="#cb7-4"></a>    <span class="dt">annotation =</span> ann,</span>
<span id="cb7-5"><a href="#cb7-5"></a>    <span class="dt">RRAalphaCutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb7-6"><a href="#cb7-6"></a>    <span class="dt">permutations =</span> <span class="dv">1000</span>,</span>
<span id="cb7-7"><a href="#cb7-7"></a>    <span class="dt">scoring =</span> <span class="st">&quot;combined&quot;</span>, </span>
<span id="cb7-8"><a href="#cb7-8"></a>    <span class="dt">permutation.seed =</span> <span class="dv">2</span></span>
<span id="cb7-9"><a href="#cb7-9"></a>  )</span></code></pre></div>
<p>Optionally, just load an example results object for testing purposes (trimming out reads as necessary)</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">data</span>(<span class="st">&quot;fit&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="kw">data</span>(<span class="st">&quot;resultsDF&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3"></a></span>
<span id="cb8-4"><a href="#cb8-4"></a>fit &lt;-<span class="st"> </span>fit[(<span class="kw">row.names</span>(fit) <span class="op">%in%</span><span class="st"> </span><span class="kw">row.names</span>(ann)),]</span>
<span id="cb8-5"><a href="#cb8-5"></a>resultsDF &lt;-<span class="st"> </span>resultsDF[(<span class="kw">row.names</span>(resultsDF) <span class="op">%in%</span><span class="st"> </span><span class="kw">row.names</span>(ann)),]</span></code></pre></div>
<p>Crispr-specific quality control and visualization tools (see man pages for details):</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="kw">ct.alignmentChart</span>(aln, sk)</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="kw">ct.rawCountDensities</span>(es, sk)</span></code></pre></div>
<p>Visualize gRNA abundance distributions</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">ct.gRNARankByReplicate</span>(es, sk) </span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw">ct.gRNARankByReplicate</span>(es, sk, <span class="dt">annotation =</span> ann, <span class="dt">geneSymb =</span> <span class="st">&quot;NoTarget&quot;</span>)  <span class="co">#Show locations of NTC gRNAs</span></span></code></pre></div>
<p>Visualize control guide behavior across conditions</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="kw">ct.viewControls</span>(es, ann, sk, <span class="dt">normalize =</span> <span class="ot">FALSE</span>)</span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="kw">ct.viewControls</span>(es, ann, sk, <span class="dt">normalize =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p>Visualize GC bias across samples, or within an experimental contrast</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="kw">ct.GCbias</span>(es, ann, sk)</span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="kw">ct.GCbias</span>(fit, ann, sk)</span></code></pre></div>
<p>View most variable gRNAs/Genes (as % of sequencing library)</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw">ct.stackGuides</span>(es,</span>
<span id="cb13-2"><a href="#cb13-2"></a>               sk,</span>
<span id="cb13-3"><a href="#cb13-3"></a>               <span class="dt">plotType =</span> <span class="st">&quot;gRNA&quot;</span>,</span>
<span id="cb13-4"><a href="#cb13-4"></a>               <span class="dt">annotation =</span> ann,</span>
<span id="cb13-5"><a href="#cb13-5"></a>               <span class="dt">nguides =</span> <span class="dv">40</span>)</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="kw">ct.stackGuides</span>(es, </span>
<span id="cb14-2"><a href="#cb14-2"></a>               sk, </span>
<span id="cb14-3"><a href="#cb14-3"></a>               <span class="dt">plotType =</span> <span class="st">&quot;Target&quot;</span>, </span>
<span id="cb14-4"><a href="#cb14-4"></a>               <span class="dt">annotation =</span> ann)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="kw">ct.stackGuides</span>(es,</span>
<span id="cb15-2"><a href="#cb15-2"></a>               sk,</span>
<span id="cb15-3"><a href="#cb15-3"></a>               <span class="dt">plotType =</span> <span class="st">&quot;Target&quot;</span>,</span>
<span id="cb15-4"><a href="#cb15-4"></a>               <span class="dt">annotation =</span> ann,</span>
<span id="cb15-5"><a href="#cb15-5"></a>               <span class="dt">subset =</span> <span class="kw">names</span>(sk)[<span class="kw">grep</span>(<span class="st">&#39;Expansion&#39;</span>, sk)])</span></code></pre></div>
<p>View a CDF of genes/guides</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="kw">ct.guideCDF</span>(es, sk, <span class="dt">plotType =</span> <span class="st">&quot;gRNA&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="kw">ct.guideCDF</span>(es, sk, <span class="dt">plotType =</span> <span class="st">&quot;Target&quot;</span>, <span class="dt">annotation =</span> ann)</span></code></pre></div>
<p>View top enriched/depleted candidates</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="kw">ct.topTargets</span>(fit,</span>
<span id="cb17-2"><a href="#cb17-2"></a>              resultsDF,</span>
<span id="cb17-3"><a href="#cb17-3"></a>              ann,</span>
<span id="cb17-4"><a href="#cb17-4"></a>              <span class="dt">targets =</span> <span class="dv">10</span>,</span>
<span id="cb17-5"><a href="#cb17-5"></a>              <span class="dt">enrich =</span> <span class="ot">TRUE</span>)</span>
<span id="cb17-6"><a href="#cb17-6"></a><span class="kw">ct.topTargets</span>(fit,</span>
<span id="cb17-7"><a href="#cb17-7"></a>              resultsDF,</span>
<span id="cb17-8"><a href="#cb17-8"></a>              ann,</span>
<span id="cb17-9"><a href="#cb17-9"></a>              <span class="dt">targets =</span> <span class="dv">10</span>,</span>
<span id="cb17-10"><a href="#cb17-10"></a>              <span class="dt">enrich =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
<p>View the gRNA behavior of gRNAs targeting a particular gene of interest</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="kw">ct.viewGuides</span>(<span class="st">&quot;Target1633&quot;</span>, fit, ann)</span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="kw">ct.gRNARankByReplicate</span>(es, sk, <span class="dt">annotation =</span> ann, <span class="dt">geneSymb =</span> <span class="st">&quot;Target1633&quot;</span>)</span></code></pre></div>
<p>View ontological enrichment within the depleted/enriched targets</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>enrichmentResults &lt;-</span>
<span id="cb19-2"><a href="#cb19-2"></a><span class="st">  </span><span class="kw">ct.PantherPathwayEnrichment</span>(</span>
<span id="cb19-3"><a href="#cb19-3"></a>    resultsDF,</span>
<span id="cb19-4"><a href="#cb19-4"></a>    <span class="dt">pvalue.cutoff =</span> <span class="fl">0.01</span>,</span>
<span id="cb19-5"><a href="#cb19-5"></a>    <span class="dt">enrich =</span> <span class="ot">TRUE</span>,</span>
<span id="cb19-6"><a href="#cb19-6"></a>    <span class="dt">organism =</span> <span class="st">&#39;mouse&#39;</span></span>
<span id="cb19-7"><a href="#cb19-7"></a>  )</span></code></pre></div>
<p>Test a gene set for enrichment within target candidates</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="kw">data</span>(<span class="st">&quot;essential.genes&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;gCrisprTools&quot;</span>)</span>
<span id="cb20-2"><a href="#cb20-2"></a>ROCs &lt;-<span class="st"> </span><span class="kw">ct.ROC</span>(resultsDF, essential.genes, <span class="dt">stat =</span> <span class="st">&quot;deplete.p&quot;</span>)</span>
<span id="cb20-3"><a href="#cb20-3"></a>PRCs &lt;-<span class="st"> </span><span class="kw">ct.PRC</span>(resultsDF, essential.genes, <span class="dt">stat =</span> <span class="st">&quot;deplete.p&quot;</span>)</span></code></pre></div>
<p>Make reports in a directory of interest</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>path2report &lt;-<span class="st">      </span><span class="co">#Make a report of the whole experiment</span></span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="st">  </span><span class="kw">ct.makeReport</span>(<span class="dt">fit =</span> fit, </span>
<span id="cb21-3"><a href="#cb21-3"></a>                <span class="dt">eset =</span> es, </span>
<span id="cb21-4"><a href="#cb21-4"></a>                <span class="dt">sampleKey =</span> sk, </span>
<span id="cb21-5"><a href="#cb21-5"></a>                <span class="dt">annotation =</span> ann, </span>
<span id="cb21-6"><a href="#cb21-6"></a>                <span class="dt">results =</span> resultsDF, </span>
<span id="cb21-7"><a href="#cb21-7"></a>                <span class="dt">aln =</span> aln, </span>
<span id="cb21-8"><a href="#cb21-8"></a>                <span class="dt">outdir =</span> <span class="st">&quot;.&quot;</span>) </span>
<span id="cb21-9"><a href="#cb21-9"></a></span>
<span id="cb21-10"><a href="#cb21-10"></a>path2QC &lt;-<span class="st">          </span><span class="co">#Or one focusing only on experiment QC</span></span>
<span id="cb21-11"><a href="#cb21-11"></a><span class="st">  </span><span class="kw">ct.makeQCReport</span>(es, </span>
<span id="cb21-12"><a href="#cb21-12"></a>                  <span class="dt">trim =</span> <span class="dv">1000</span>, </span>
<span id="cb21-13"><a href="#cb21-13"></a>                  <span class="dt">log2.ratio =</span> <span class="fl">0.05</span>, </span>
<span id="cb21-14"><a href="#cb21-14"></a>                  <span class="dt">sampleKey =</span> sk, </span>
<span id="cb21-15"><a href="#cb21-15"></a>                  <span class="dt">annotation =</span> ann, </span>
<span id="cb21-16"><a href="#cb21-16"></a>                  <span class="dt">aln =</span> aln, </span>
<span id="cb21-17"><a href="#cb21-17"></a>                  <span class="dt">identifier =</span> <span class="st">&#39;Crispr_QC_Report&#39;</span>,</span>
<span id="cb21-18"><a href="#cb21-18"></a>                  <span class="dt">lib.size =</span> <span class="ot">NULL</span></span>
<span id="cb21-19"><a href="#cb21-19"></a>                  )                </span>
<span id="cb21-20"><a href="#cb21-20"></a></span>
<span id="cb21-21"><a href="#cb21-21"></a>path2Contrast &lt;-<span class="st">    </span><span class="co">#Or Contrast-specific one</span></span>
<span id="cb21-22"><a href="#cb21-22"></a><span class="st">  </span><span class="kw">ct.makeContrastReport</span>(<span class="dt">eset =</span> es, </span>
<span id="cb21-23"><a href="#cb21-23"></a>                        <span class="dt">fit =</span> fit, </span>
<span id="cb21-24"><a href="#cb21-24"></a>                        <span class="dt">sampleKey =</span> sk, </span>
<span id="cb21-25"><a href="#cb21-25"></a>                        <span class="dt">results =</span> resultsDF, </span>
<span id="cb21-26"><a href="#cb21-26"></a>                        <span class="dt">annotation =</span> ann, </span>
<span id="cb21-27"><a href="#cb21-27"></a>                        <span class="dt">comparison.id =</span> <span class="ot">NULL</span>, </span>
<span id="cb21-28"><a href="#cb21-28"></a>                        <span class="dt">identifier =</span> <span class="st">&#39;Crispr_Contrast_Report&#39;</span>)            </span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
